node/core/runtime-api/src/lib.rs

```diff
@@ -342,8 +349,65 @@ where
 		Request::CheckValidationOutputs(para, commitments, sender) =>
 			query!(CheckValidationOutputs, check_validation_outputs(para, commitments), sender),
 		Request::SessionIndexForChild(sender) => query!(SessionIndexForChild, session_index_for_child(), sender),
-		Request::ValidationCode(para, assumption, sender) =>
-			query!(ValidationCode, validation_code(para, assumption), sender),
+		Request::ValidationCode(para, assumption, sender) => {
```

Comment:
This the only calls to the runtime parachain host API `validate_code` that I could found. So there shouldn't be any other need to support both old and new version in the codebase as far as I can see.

Reply:
Yes, all runtime API request go through runtime api subsystem.